packages/main/src/TimePicker.ts

```diff
@@ -705,6 +748,32 @@ class TimePicker extends UI5Element implements IFormInputElement {
 		return TimePicker.i18nBundle.getText(TIMEPICKER_CANCEL_BUTTON);
 	}
 
+	get hasValueStateText(): boolean {
+		return this.hasValueState && this.valueState !== ValueState.Positive;
+	}
+
+	get hasValueState(): boolean {
```

Comment:
I'm confused with this and the above function, should clean up probably? what is the difference